apps/examples/smartfs/buffer_optimization/buffer_optimization_main.c

```diff
@@ -0,0 +1,466 @@
+{
+	printf("\n**************************************************************************************\n");
+	printf("Throughput while %s %dKb file %d time(s) with %dbyte buffer size = %.02fKbps\n",\
+			name, size, itr, buf_size, (float)(size * 1000000) / time);
+	printf("\n**************************************************************************************\n");
+	return;
+}
+
+/****************************************************************************
+ * Name: create_test_file
+ *
+ * Description: Creates the test file with test data that we will use
+ *              to conduct the test.
+ *
+ ****************************************************************************/
+
+static int create_test_file(int f_size)
+{
+	int fd;
+	int ret = OK;
+	memset(g_buf, 1024, '1');
+
+	fd = open(SAMPLE_FILE, O_WROK | O_CREAT);
+	if (fd < 0) {
+		printf("Unable to open Sample File: %s for testing, Error: %d\n", SAMPLE_FILE, errno);
+		return ERROR;
+	}
+
+	while (f_size >= 1024) {
+		ret = write(fd, g_buf, 1024);
+		if (ret != 1024) {
+			printf("Unable to write to Sample File %s, Error: %d\n", SAMPLE_FILE, errno);
+			ret = errno;
+			goto error_with_fd;
+		}
+		f_size -= 1024;
+	}
+	if (f_size) {
+		ret = write(fd, g_buf, f_size);
+		if (ret != f_size) {
+			printf("Unable to write to Sample File %s, Error: %d\n", SAMPLE_FILE, errno);
+			ret = errno;
+			goto error_with_fd;
+		}
+	}
+	ret = OK;
+
+error_with_fd:
+	close(fd);
+	return ret;
+}
+
+/****************************************************************************
+ * Name: append_test
+ *
+ * Description: Perform appending of data to the file, buffer by buffer
+ *
+ ****************************************************************************/
+
+static int append_test(int size)
+{
+	int ret = OK;
+	int fd;
+	int itr = 0;
+	int total = TOTAL_APPEND_SIZE;
+
+	while (itr < APPEND_TEST_ITR) {
+		fd = open(SAMPLE_FILE, O_WROK | O_APPEND);
+		if (fd < 0) {
+			printf("Unable to open sample file %s for appending, Error: %d\n", SAMPLE_FILE, errno);
+			return ERROR;
+		}
+
+		while (total >= size) {
+			ret = write(fd, g_buf, size);
+			if (ret != size) {
+				printf("Unable to write to file %s, ret: %dbytes, Error: %d\n", SAMPLE_FILE, ret, errno);
+				ret = errno;
+				goto error_with_fd;
+			}
+			total -= size;
+		}
+		if (total) {
+			ret = write(fd, g_buf, total);
+			if (ret != total) {
+				printf("Unable to write to file %s, ret: %dbytes, Error: %d\n", SAMPLE_FILE, ret, errno);
+				ret = errno;
+				goto error_with_fd;
+			}
+		}
+		close(fd);
+		itr += 1;
+	}
+
+	ret = OK;
+error_with_fd:
+	close(fd);
+	return ret;
+}
+
+/****************************************************************************
+ * Name: append_test_util
+ *
+ * Description: Set up the timer and run the append test for different
+ *              buffer sizes.
+ *
+ ****************************************************************************/
+
+static int append_test_util(int frt_fd, int size)
```

Comment:
append_test -> append_file or do_append
append_test_util -> append_test or something.
Util is correct naming for some kind of helper...